### PR TITLE
Add shadow

### PR DESCRIPTION
**Pull request by at-wat**
- [x] after #54 
- [x] after #77

### CODE REVIEW

**Comment by at-wat** [comment]
@kt3k @t-asaka please take a look

File: shadow/state.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	if err := updateStateRaw(s.State.Desired, state.State.Desired); err != nil {
+		return err
+	}
+	if err := updateStateRaw(s.State.Reported, state.State.Reported); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *ThingDocument) updateDelta(state *thingDelta) (bool, error) {
```
Comment (kt3k) [inline shadow/state.go:59]: Does this function need to return (bool, error)? (There seems no path for returning err != nil

Comment (at-wat) [reply shadow/state.go]: Removed error from return values.

**Comment by at-wat** [comment]
@kt3k PTAL

**Pull request merged**